pkg/asset/agent/image/agentpxefiles.go

```diff
@@ -118,6 +131,51 @@ func (a *AgentPXEFiles) PersistToFile(directory string) error {
 		return err
 	}
 
+	tmpdir, err := os.MkdirTemp("", pxeAssetsPath)
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(tmpdir)
+
+	fileReader, err := isoeditor.GetFileFromISO(a.isoPath, "coreos/kargs.json")
+	if err != nil {
+		return err
+	}
+	defer fileReader.Close()
+
+	data, err := ioutil.ReadAll(fileReader)
+	if err != nil {
+		return err
+	}
+
+	var configFile CoreOSKargs
+	json.Unmarshal(data, &configFile)
+
+	var file filesystem.File
```

Comment:
I'm not sure that this whole part is really needed: the `default` field of `kargs.json` already contains the desired data (except for the `coreos.liveiso`)

Reply:
Discussed and confirmed parsing the default value from kargs.json